### DISCUSSION TITLE
Collecting page data error after NextJS version 8 to version 10 upgrade

### DISCUSSION THREAD

**Discussion by chrischandler25** [vercel/next.js#37579] [Help] [answered]
I asked a version similar question at [https://github.com/vercel/next.js/discussions/35534](url) but don't think I'll receive any response as it's an old thread.

I inherited an app that is based on NextJS.  It seems to me that NextJS was not the best choice of framework as the app is internal to our organisation, is entirely dynamic and has no use for SSR.  I do intend to move the app out of NextJS, but this isn't an option in the short term.

I updated the version of Next from 8 to 10 in order to update some other required dependencies.  I made the changes needed to deliver a new feature which all worked well on my local dev server, however when I pushed to CI which uses "npm run build", I discovered that the build process is very different to the previous version.  It seems that it now attempts to generate HTML as part of the build for all of the pages, and is failing at a page which unfortunately isn't named in the error.  I'm also finding it very hard to trace the source of the error as it's named in one of the chunk JS files which is webpack output from the build and very hard to follow.  I have copied the error at the bottom of this post.

I've tried to find a way to disable this process, however haven't been able to come up with anything.  Really I'm just looking for a solution to deploy while I move this application out of NextJS.  

Any help would be greatly appreciated

------------------------------

info - Collecting page data ...node:internal/process/promises:279
triggerUncaughtException(err, true /* fromPromise */);
^

TypeError: Cannot read properties of undefined (reading 'N5')
at Object.N5 (C:\dev\alvitas\UI.next\server\chunks\966.js:794:80)
at Object.1411 (C:\dev\alvitas\UI.next\server\chunks\966.js:5363:51)
at webpack_require (C:\dev\alvitas\UI.next\server\webpack-runtime.js:25:42)
at Object.6772 (C:\dev\alvitas\UI.next\server\chunks\966.js:7107:14)
at webpack_require (C:\dev\alvitas\UI.next\server\webpack-runtime.js:25:42)
at Object.8291 (C:\dev\alvitas\UI.next\server\chunks\966.js:9567:64)
at webpack_require (C:\dev\alvitas\UI.next\server\webpack-runtime.js:25:42)
at Object.2914 (C:\dev\alvitas\UI.next\server\chunks\966.js:8290:73)
at webpack_require (C:\dev\alvitas\UI.next\server\webpack-runtime.js:25:42)
at Object.5966 (C:\dev\alvitas\UI.next\server\chunks\966.js:8778:15) {
type: 'TypeError'
}

**Comment by icyJoseph** [comment, marked as answer]
I recently learned that even if `getInitialProps` is enabled at _app level, `Next.js` still tries to pre-render pages that are somewhat static, meaning that don't use `getInitialProps` or `getServerSideprops`.

Given that you seem to want to abandon Next.js perhaps, try first to define a dummy `gSSP` on each page:

```js
export const getServerSideProps = () => {
  return {props: {}}
}
```

That should really force things to be rendered on demand, but still, I am not sure how much it'll help, **because it might just be moving the issue to the runtime.**

A better solution would be to identify which import/module, is using code that would only run on Node.js, but is being packed to be used in the browser. One thing you could try, is to move all but one page to another folder, run build, if it goes well, add a new page, and repeat, until one of the pages makes the build crash. Of course, assuming the build also crashes locally.

**Reply by chrischandler25** [reply]
Thanks so much for this - it put me very much in the right direction.  

I moved all pages until I found the offending page (_app!), and then used a process of elimination to find the offending component.  It turned out to be a simple loading spinner, and I have not a clue why it caused an issue as it's used in many places and seems quite valid.  To solve the problem, I simply copied the spinner definition into my _app.tsx and it worked.  I thought this may be related to module loading, so tested an alternative component from the same location and yet that worked.  In this instance, I'm happy to live with good enough as it's a short term fix. 

I did also need to add getServerSideProps to all of the other pages, as they had their own set of issues.  The good news is however, I haven't seen any runtime issues post deployment.

Thanks again!
